homeassistant/components/homekit/__init__.py

```diff
@@ -214,6 +217,10 @@ def add_bridge_accessory(self, state):
         """Try adding accessory to bridge if configured beforehand."""
         if not state or not self._filter(state.entity_id):
             return
+        self.devices_added += 1
+        if self.devices_added == MAX_DEVICES + 1:
+            _LOGGER.warning("More than %s devices added to homekit.",
+                            MAX_DEVICES)
```

Comment:
What do you think of adding additional information about:
- what could happen (HomeKit doesn't work as expected)
- what to do (consider using the filter option)

Reply:
@cdce8p  The impact if the error is quite intense - nothing works and apple tv refuses to pair without any further warning - I would personally opt for a more drastic step. e.g. shrinking the devices to  MAX_DEVICES + the warning.

It took me about 2 hours to find out what is wrong.

Reply:
This could break some setup unintentionally. Apple just strongly recommends to use 100 or less accessories per Bridge but I have seen cases with more than that work without issues.

It's also listed in the documentation: https://www.home-assistant.io/components/homekit/#device-limit